tests/geneset.py

```diff
@@ -0,0 +1,259 @@
+        self.assertEqual(gs_li[2]['_id'], 'WP103')
+
+    def test_query(self):
+        qres = self.mgs.query('genes.mygene_id:2937', size=5)
+        self.assertTrue('hits' in qres)
+        self.assertEqual(len(qres['hits']), 5)
+
+    def test_query_default_fields(self):
+        self.mgs.query(q='glucose')
+
+    def test_query_field(self):
+        self.mgs.query(q='genes.ncbigene:1017')
+
+    def test_species_filter_plus_query(self):
+        dog = self.mgs.query(q='glucose', species='9615')
+        self.assertEqual(dog['hits'][0]['taxid'], "9615")
+
+    def test_query_by_id(self):
+        query = self.mgs.query(q="_id:WP100")
+        self.assertEqual(query['hits'][0]['_id'], "WP100")
+
+    def test_query_by_name(self):
+        kinase = self.mgs.query(q='name:kinase')
+        self.assertIn('kinase', kinase['hits'][0]['name'].lower())
+
+    def test_query_by_description(self):
+        desc = self.mgs.query(q='description:cytosine deamination')
+        self.assertIn('cytosine', desc['hits'][0]['description'].lower())
+        self.assertIn('deamination', desc['hits'][0]['description'].lower())
+
+    def test_query_by_source_go(self):
+        go = self.mgs.query(q='source:go', fields='all')
+        self.assertIn('go', go['hits'][0].keys())
+        self.assertEqual(go['hits'][0]['source'], 'go')
+
+    def test_query_by_source_ctd(self):
+        ctd = self.mgs.query(q='source:ctd', fields='all')
+        self.assertIn('ctd', ctd['hits'][0].keys())
+        self.assertEqual(ctd['hits'][0]['source'], 'ctd')
+
+    def test_query_by_source_msigdb(self):
+        msigdb = self.mgs.query(q='source:msigdb', fields='all')
+        self.assertIn('msigdb', msigdb['hits'][0].keys())
+        self.assertEqual(msigdb['hits'][0]['source'], 'msigdb')
+
+    @pytest.mark.skip(reason="Has to configure credentials")
+    def test_query_by_source_kegg(self):
+        kegg = self.mgs.query(q='source:kegg', fields='all')
+        self.assertIn('kegg', kegg['hits'][0].keys())
+        self.assertEqual(kegg['hits'][0]['source'], 'kegg')
+
+    def test_query_by_source_do(self):
+        do = self.mgs.query(q='source:do', fields='all')
+        self.assertIn('do', do['hits'][0].keys())
+        self.assertEqual(do['hits'][0]['source'], 'do')
+
+    def test_query_by_source_reactome(self):
+        reactome = self.mgs.query(q='source:reactome', fields='all')
+        self.assertIn('reactome', reactome['hits'][0].keys())
+        self.assertEqual(reactome['hits'][0]['source'], 'reactome')
+
+    def test_query_by_source_smpdb(self):
+        smpdb = self.mgs.query(q='source:smpdb', fields='all')
+        self.assertIn('smpdb', smpdb['hits'][0].keys())
+        self.assertEqual(smpdb['hits'][0]['source'], 'smpdb')
+
+    @unittest.skipIf(not biothings_client.caching_avail, "requests_cache not available")
+    def test_caching(self):
+        def _getgeneset():
+            return self.mgs.getgenesets("1017")
```

Comment:
`1017`, `1018` are not good ids for geneset, use something like `WP100` or other values from `_id` field